cfpb/consumerfinance.gov
# Setup custom signal processing for Elasticsearch

Since we want our Elasticsearch indices to follow the behavior of wagtail pages, and not django models it makes sense for us to create a custom signal handler and hook into wagtail signals instead of relying on the django signals for our events. 


---



## Additions

- `WagtailSignalProcessor` - hooks into the wagtail publish/unpublish and move events to allow us to handle index updating based on page events that mirror what actually occurs within our site. 

## Changes

- Now use the new signal processor for handling index events.


## How to test this PR

1. Create a new page - Like a blog post but save it as a draft
2. Ensure the page does not exist in the index
3. Publish page and confirm it now appears in the index
4. Unpublish page and confirm it no longer appears in the index
5. Move a published page to the Trash site and confirm it is deleted from the index

## Notes and todos

- 


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance
